#159340982 add user signup

damafeez opened the pull request:
#### What does this PR do?

- add user signup controller
- add P0ST /auth/signup route
- add passing POST /auth/signup test

#### Description of Task to be completed?
- password hashing
- more test cases

#### How should this be manually tested?
run `npm test`

#### Any background context you want to provide?
n/a

#### What are the relevant pivotal tracker stories?
#159340982

Questions:
None

damafeez committed:
feature(user): add user signup

- add user signup controller
- add user signup test

[Working on 159340982]

damafeez committed:
feature(user): add user signup

- add user signup controller
- add user signup test

[Working on 159340982]

fidelisojeah commented on server/controllers/user.js:
```diff
@@ -0,0 +1,33 @@
+  try {
+    const { fullName, username, password } = req.body;
+    if (fullName && username && password) {
+      const authenticationQuery = `INSERT INTO authentication (username, password)
```
You should also insert into the users table as well

fidelisojeah reviewed (commented):
Include the postgreSQL module in your travis: 
```yml
services:
  - postgresql
```
Increase your test coverage also

damafeez committed:
feature(user): add user signup

- add failing signupSpec test

[Working on 159340982]

damafeez committed:
feature(user): add user signup

- complete user signup
- signup test passing

    [Working on 159340982]

fidelisojeah commented on server/models/User.js:
```diff
@@ -0,0 +1,79 @@
+  }
+
+  async save() {
+    const authQuery = 'INSERT INTO authentication(username, password) VALUES($1, $2) RETURNING *';
```
Okay rather than just sending the password directly into the database, you should hash it
Good job though

damafeez replied:
Yes I will. It's added in the `Description of Task to be completed?`. Thanks.

damafeez committed:
feature(user): add user signup

    -  drop mydiary_test db on Travis before creating new one

    [Working on 159340982]
